pkg/signing/plugins/cosign/cosign.go

```diff
@@ -0,0 +1,99 @@
+package cosign
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"get.porter.sh/porter/pkg/portercontext"
+	"get.porter.sh/porter/pkg/signing/plugins"
+	"get.porter.sh/porter/pkg/tracing"
+)
+
+var _ plugins.SigningProtocol = &Cosign{}
+
+// Signer implements an in-memory signer for testing.
+type Cosign struct {
+	PublicKey        string
+	PrivateKey       string
+	RegistryMode     string
+	Experimental     bool
+	InsecureRegistry bool
+}
+
+func NewSigner(c *portercontext.Context, cfg PluginConfig) *Cosign {
+
+	s := &Cosign{
+		PublicKey:        cfg.PublicKey,
+		PrivateKey:       cfg.PrivateKey,
+		RegistryMode:     cfg.RegistryMode,
+		Experimental:     cfg.Experimental,
+		InsecureRegistry: cfg.InsecureRegistry,
+	}
+
+	return s
+}
+
+// TODO: we should get the certificate... here?
+func (s *Cosign) Connect(ctx context.Context) error {
```

Comment:
cosign requires the path to the private key to be passed on the command line when signing, and the path to the public key when verifying, so I don't think it makes sense to get any of the keys at the moment.

Regarding you point about checking if the tool is installed, I think that would be a great idea, I will add that.
I don't think we should install it automatically, as it could be a source of confusion, especially if people later want to install it in another version later, but for sure an error should be thrown if it isn't available.